### PR TITLE
Add identicon.js@2.1.0 w/ npm auto-update

### LINKED ISSUE #10256: [Request] Add identicon.js
**Library name:** identicon.js
**Git repository url:** https://github.com/stewartlord/identicon.js

### PR DESCRIPTION
**Pull request by extend1994**
Pull request for issue: #10256

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/stewartlord/identicon.js
 * Official website (optional, not the repository): 
 * NPM package url (optional): https://www.npmjs.com/package/identicon.js
 * License and its reference: [BSD-2-Clause](https://github.com/stewartlord/identicon.js#usage)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 8
   - Count of stars: 287
   - Count of forks: 52
 * NPM download stats (optional):
   - Downloads in the last day: 42
   - Downloads in the last week: 296
   - Downloads in the last month: 794

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #10256, cc @stewartlord

### CODE REVIEW

**Comment by extend1994** [comment]
@PeterDaveHello please help me review this pull request, thank you.

**Comment by extend1994** [comment]
@PeterDaveHello I modified the auto-update source from git to npm, please help me review this, thank you.

**Comment by extend1994** [comment]
@PeterDaveHello I modified the auto-update source from git to npm, please help me review this, thank you.

**Comment by maruilian11** [comment]
where is the npm info?

File: ajax/libs/identicon.js/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "identicon"
+  ],
+  "author": "stewardlord",
+  "license": "BSD-2-Clause",
+  "npmName": "identicon.js",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "identicon*.+(js|css|map)",
```
Comment (maruilian11) [inline ajax/libs/identicon.js/package.json]: `"identicon*",`
`"pnglib*"`

Comment (PeterDaveHello) [inline ajax/libs/identicon.js/package.json]: It's no harm to use `.+(js|css|map)` by the way.

Comment (maruilian11) [inline ajax/libs/identicon.js/package.json]: ok

Comment (maruilian11) [inline ajax/libs/identicon.js/package.json]: then, just need to update the version and commit msg. thanks!

**Comment by extend1994** [comment]
@PeterDaveHello @maruilian11 I didn't notice the current version is different from one when I publish the first pull request, sorry for that. I updated the all files, please help me review it, thank you.

File: ajax/libs/identicon.js/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        "pnglib*.+(js|css|map)"
```
Comment (pvnr0082t) [inline ajax/libs/identicon.js/package.json]: These two can be combined into `"+(identicon|pnglib)*.+(js|css|map)"`.

**Comment by extend1994** [comment]
@maruilian11 please help me view this pull request, thank you.
@pvnr0082t please help me view this pull request. I want to ask what's the difference between them? Just being shorter or ...? Thank you.

**Comment by maruilian11** [comment]
where is the npm info in the init comment?

**Comment by extend1994** [comment]
@maruilian11 Sorry for forgetting to do that, I just updated it.

**Comment by maruilian11** [comment]
LGTM

**Comment by pvnr0082t** [comment]
@extend1994 Yes. We'll combined the rules if we can improve them and still not so complicated.

**Comment by PeterDaveHello** [comment]
Your license reference is wrong...

**Pull request merged**